### ISSUE TITLE
kubectl-version hint is misleading/confusing

### ISSUE THREAD

**Issue by philomory** [linkerd/linkerd2#9615]
When I run `linkerd check` from a machine with `linkerd` 2.12.1 and `kubectl` 1.20.4 against a cluster running k8s 1.21.14, I get the following output:

```
× is running the minimum kubectl version
    kubectl is on version [1.20.4], but version [1.21.0] or more recent is required
    see https://linkerd.io/2.12/checks/#kubectl-version for hints
```

However, when I follow that link, the "hint" says the following:
>Linkerd requires at least version 1.13.0. Verify your kubectl version with:
>
> ```
> kubectl version --client --short
> ```
> To fix please update kubectl version.
> For more information on upgrading Kubernetes, see the page in the Kubernetes Documentation.

And not much more.

This is at best misleading. The troubleshooting tip should either explain that `linkerd` requires your local `kubectl` version to be at least as high as your cluster's major version (contrary to `kubectl`'s own requirements, which is simply to be within one major version in either direction), or should state that `linkerd` requires your local `kubectl` version to be at least `1.21.0` regardless of Kubernetes version, or otherwise indicate the actual requirement for passing this check (whatever that actually is).

**Comment by dtaskai** [comment]
@adleong I'll be away for next week, but after that I would love to work on this one!

**Comment by alpeb** [comment]
Fixed by #9623
